dashboard/app/lib/javascripts/dashboard/main.js

```diff
@@ -111,11 +172,22 @@ window.Dashboard = {
 			break;
 
 			case "SERVICE_UNAVAILABLE":
-				if (window.location.protocol === "http:" && this.config.INSTALL_CERT) {
-					this.__handleConfigReady();
-				} else {
+//				if (window.location.protocol === "http:" && this.config.INSTALL_CERT) {
+//					this.__handleConfigReady();
+//				} else {
 					this.__handleServiceUnavailable(event.status);
-				}
+//				}
+
+			case "CONTROLLER_UNAVAILABLE_FROM_HTTPS":
+				console.log("CONTROLLER_UNAVAILABLE_FROM_HTTPS", window.location.protocol);
+				// Controller isn't accessible via https. Redirect to http and try again.
+				window.location.href = window.location.href.replace("https", "http");
+				break;
+
+			case "HTTPS_CERT_MISSING":
+				console.log("HTTPS_CERT_MISSING", window.location.protocol);
+				Marbles.history.navigate("installcert");
+				Marbles.history.loadURL();
```

Comment:
No need to call `loadURL` here as `run` has already been called (the `loadURL` calls in `run` are to load the initial route due to the `trigger: false` option passed to `Marbles.history.start`, though the one called directly after `navigate` isn't necessary as `trigger: false` only effects the initial route).